### PR TITLE
remove compile features from std.experimental.logger

### PR DESCRIPTION
**Pull request by burner**
@atilaneves as promised

### CODE REVIEW

File: changelog/logger_remove_compile_time_loglevel.dd [hunk 1]
```diff
@@ -0,0 +1,9 @@
+remove std.experimental.logger's capability to set the minimal LogLevel at
+compile time
```
[the remark this answers is not in the document]
Comment (burner) [reply changelog/logger_remove_compile_time_loglevel.dd]: I was able to fix this ;-)

**Review by atilaneves — APPROVED** [review]

**Comment by burner** [comment]
@ibuclaw @atilaneves thank you

File: changelog/logger_remove_compile_time_loglevel.dd [hunk 2]
```diff
@@ -0,0 +1,8 @@
+remove std.experimental.logger's capability to set the minimal LogLevel at compile time
+
+Before this change std.experimental.logger had the capability to disable
+logging at compile time.
+It was also possible to set the minimal LogLevel at compile time.
+The trade-off between gained capability, added complexity, and error-proneness
+was too heavily tilted the second two items.
```
Comment (RazvanN7) [inline changelog/logger_remove_compile_time_loglevel.dd:7]:
```suggestion
was too heavily tilted towards the second two items.
```

File: std/experimental/logger/core.d [hunk 3]
```diff
@@ -12,56 +12,6 @@ import std.traits;
...
-The previously described version statements are used to decide if the
-`LogLevel` is active. The version statements only influence the compile
-unit they are used with, therefore this function can only disable logging this
-specific compile unit.
-*/
-template isLoggingActiveAt(LogLevel ll)
-{
-    version (StdLoggerDisableLogging)
-    {
-        enum isLoggingActiveAt = false;
```
Comment (RazvanN7) [inline std/experimental/logger/core.d:25]: shouldn't we issue a deprecation when this version is used?

Comment (ibuclaw) [reply std/experimental/logger/core.d:25]: It's an experimental module.

Comment (burner) [reply std/experimental/logger/core.d:25]: what Iain said

File: changelog/logger_remove_compile_time_loglevel.dd (hunk 2 above)
Comment (burner) [reply changelog/logger_remove_compile_time_loglevel.dd]: thank you

**Comment by burner** [comment]
thank you all

**Pull request merged**